### PR TITLE
Refactor api.shouldDo and fix https://github.com/HabitRPG/habitrpg/issues/5395

### LINKED ISSUE #5395: Daily Start Date not working correctly for new Dailies when a Custom Day Start is set
I'm noticing on my account with a CDS of 5 that when I create a Daily with a Start Date of today and "Every X Days" greater than 1, that the Daily is not due today. If the number of days is 1, it is due today. My timezone is GMT+10.

The same thing has been reported by another user (987c6d4f-72f2-49a7-888a-e853b11eaff7) with a CDS of 4. GMT+12.

It does not happen on my other account with a CDS of 0 (the Daily is due today when it has a Start Date of today and "Every X Days" greater than or equal to 1). GMT+10.

### PR DESCRIPTION
**Pull request by Alys**
**Do not merge yet!**   I'm posting now for a code review.

This fixes https://github.com/HabitRPG/habitrpg/issues/5395 (Dailies weren't being shown/hidden correctly because of Start Date - CDS interaction). This might have been just a visual problem on the tasks screen rather than a cron problem.

It also refactors api.shouldDo. You might find it easier to follow the changes by looking at the commits one by one, although they are a bit verbose and pedantic.

### CODE REVIEW

**Comment by Alys** [comment]
I've finished testing this and I think it's okay.

Could someone please review all of my changes? It refactors api.shouldDo. You might find it easier to follow the changes by looking at the commits one by one.

File: common/script/content.coffee [hunk 1]
```diff
@@ -1919,7 +1919,7 @@ api.subscriptionBlocks =
   basic_12mo: months:12, price:48
 _.each api.subscriptionBlocks, (b,k)->b.key = k
 
-repeat = {m:true,t:true,w:true,th:true,f:true,s:true,su:true}
+# repeat = {m:true,t:true,w:true,th:true,f:true,s:true,su:true}
```
Comment (crookedneighbor) [inline common/script/content.coffee:1922]: This is commented out because dailies are not populated in the userDefaults right  now, right?

**Comment by crookedneighbor** [comment]
This refactor gets a big :+1: from me. 

It always irked me that we mixed true/false with 1/0 in the repeat. I never understood why. If it had meaning at one point, it clearly doesn't have that meaning now.

File: common/script/content.coffee (hunk 1 above)
Comment (Alys) [inline common/script/content.coffee:1922]: Not exactly. It was used only by the Dailies lines a bit below that (default Dailies for new accounts) and those lines were commented out in recent weeks, so I've commented out the `repeat` for consistency.

Comment (crookedneighbor) [inline common/script/content.coffee:1922]: Yeah, that's what I meant. Good.

I'm actually not sure this is needed at all. The defaults for a daily should be that it repeats on all days of the week.

Comment (Alys) [inline common/script/content.coffee:1922]: Ah, sorry, I actually misread your first comment.

Yes, good point about not needed.

**Comment by Alys** [comment]
I'll merge this in a few hours if no one gives reasons not to.

**Comment by crookedneighbor** [comment]
I'm :+1: on this being merged in now.

**Pull request merged**